packages/python/plotly/plotly/io/_json.py

```diff
@@ -57,6 +57,25 @@ def coerce_to_strict(const):
         return const
 
 
+_swap_json = (
+    ("<", "\\u003c"),
+    (">", "\\u003e"),
+    ("/", "\\u002f"),
+)
+_swap_orjson = _swap_json + (
+    ("\u2028", "\\u2028"),
+    ("\u2029", "\\u2029"),
+)
+
+
+def _safe(json_str, _swap):
+    out = json_str
+    for unsafe_char, safe_char in _swap:
+        if unsafe_char in out:
+            out = out.replace(unsafe_char, safe_char)
```

Comment:
Testing first whether the character is present in the string is substantially faster than `.replace` when it finds even a single instance of the character - so given that much of the time you won't have some of these characters it's worthwhile including the `if` first. And looping over each character in turn is a lot faster than any solution I could find that only searches the string once, ie a regexp.